File: ulohy/20-pozpatku/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let n;
const numbers = [];

rl.on("line", (line) => {
    if (n === undefined) {
        n = parseInt(line);
        return;
    }

    if (line === "") {
        return;
    }

    const number = parseInt(line);
    numbers.push(number);

    if (numbers.length < n) {
        return;
    }

    for (let i = n - 1; i >= 0; i--) {
        console.log(numbers[i]);
    }

    rl.close();
});

File: ulohy/20-pozpatku/reseni/main.cpp
#include <iostream>
#include <vector>

int main() {
    int n;
    std::cin >> n;

    std::vector<int> numbers(n);

    for (int i = 0; i < n; i++) {
        std::cin >> numbers[i];
    }

    for (int i = n - 1; i >= 0; i--) {
        std::cout << numbers[i] << std::endl;
    }
}
